### DISCUSSION TITLE
Is there a way for me to log in to my actual minecraft account when developing?

### DISCUSSION THREAD

**Discussion by nad2040** [FabricMC/fabric-api#1618] [Mod Dev Support] [answered]
I'm using IntelliJ and when I run the Minecraft Client configuration, how do I make it so that I can use my real MC account instead of having a random Player___ account. I heard you could use --username and --password, but I tried it and it says `Completely ignored arguments: [--password, ********]`

Running mc 1.17.1

**Comment by warjort** [comment]
Minecraft itself doesn't use a password, it uses an access token.
The password is used by the launcher to create the token.

You can replicate what the launcher does:
https://wiki.vg/Authentication

Then the parameters are:
--username
--accessToken
--uuid
--userType

The userType is either legacy or mojang depending upon what account you have.

**Reply by nad2040** [reply]
Thanks. I probably will just use the Player___ account then.

**Comment by nok-ko** [comment, marked as answer]
I recommend putting a mod like [Auth Me](https://www.curseforge.com/minecraft/mc-mods/auth-me) in your dev environment instead of messing with command-line variables. IIRC that can leak sensitive info into logs if you're not careful. (Could be very wrong though!)

**Comment by xdx505** [comment]
I have found the solution. Maybe it can be helpful for anybody.
1. Install [MultiMC ](https://multimc.org/) launcher and log in into your account in it.
2. Then go to MultiMC installation folder and find `acounts.json` file
3. You'll find your valid token in this part. **Save it for future**:
```json
"ygg": {
   "token": "your_token"
}
```
4. Do HTTP GET request to `https://api.minecraftservices.com/minecraft/profile` ([More info](https://wiki.vg/Microsoft_Authentication_Scheme#Getting_the_profile)) with Bearer header with your token. For example, via Postman, like this:

![image](https://github.com/FabricMC/fabric/assets/66822303/354f78ea-1b4d-4af7-b10e-85d2da3cf7d1)

5. **Save your id** from **first** field

![image](https://github.com/FabricMC/fabric/assets/66822303/e697c2ff-e588-44a4-ae00-67abba3a209b)

6. Edit your IDEA run configuration "Minecraft Client". Paste in "Program arguments" this:
`--username <YOUR_USERNAME> --uuid <YOUR_ID_FROM_STEP_5> --accessToken <YOUR_TOKEN_FROM_STEP_1> --userType msa`

![image](https://github.com/FabricMC/fabric/assets/66822303/0c8f38ba-55c0-45e3-86f9-c0ffbcd6115a)

7. Success!

**Comment by cputnam-a11y** [comment]
Dev auth is also a solution

**Comment by lishsite-dev** [comment]
You can use DevAuth for this. It lets you authenticate your Microsoft account through OAuth and then use that account when launching the Fabric dev client, without putting your password in the run arguments. It’s much cleaner than trying to pass --password, since [Minecraft](http://minecrftapkmod.com.br/) expects an access token rather than your account password.

**Comment by modmuss50** [comment]
Fabric Loom 1.18 has an official solution to this with the `microsoftLogin` task, there is no need to use a 3rd party tool.
